### PR TITLE
Fix WakeEvent::Open()

### PR DESCRIPTION
**Pull request by kpschoedel**
#### Problem

Some tests were reporting,
```
CHIP:CSL: System wake event notify failed: OS Error 0x02000009: Bad file descriptor
```
but not failing.

#### Change overview

- Fix the problem (setting `mWriteFD` too late in `WakeEvent::Open()`).
- Add error returns and checks so that affected tests actually fail
  without the above fix.

#### Testing

With the added error handling, but without the `WakeEvent::Open()` change,
numerous tests fail.

### CODE REVIEW

File: src/inet/TCPEndPoint.cpp [hunk 1]
```diff
@@ -1648,8 +1666,9 @@ CHIP_ERROR TCPEndPoint::DoClose(CHIP_ERROR err, bool suppressCallback)
                     ChipLogError(Inet, "SO_LINGER: %d", errno);
             }
 
-            if (mSocket.Close() != 0 && err == CHIP_NO_ERROR)
-                err = chip::System::MapErrorPOSIX(errno);
+            CHIP_ERROR status = mSocket.Close();
+            if (status != CHIP_NO_ERROR && err == CHIP_NO_ERROR)
```
Comment (bzbarsky-apple) [inline src/inet/TCPEndPoint.cpp:1670]: Could simplify to:
```suggestion
            if (err == CHIP_NO_ERROR)
```
in that if `err == CHIP_NO_ERROR` and `status == CHIP_NO_ERROR` the assignment is still fine.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

File: src/system/WatchableEventManagerSelect.h [hunk 2]
```diff
@@ -41,9 +41,9 @@ class WatchableSocket;
 class WatchableEventManager
 {
 public:
-    void Init(System::Layer & systemLayer);
-    void Shutdown();
-    void Signal();
+    CHIP_ERROR Init(System::Layer & systemLayer);
+    CHIP_ERROR Shutdown();
+    CHIP_ERROR Signal();
```
Comment (mspang) [inline src/system/WatchableEventManagerSelect.h:46]: Can you document these?

When can signalling fail?

File: src/system/WatchableEventManagerSelect.cpp [hunk 3]
```diff
@@ -87,6 +88,7 @@ void WatchableEventManager::Signal()
     {
         ChipLogError(chipSystemLayer, "System wake event notify failed: %" CHIP_ERROR_FORMAT, ChipError::FormatError(status));
```
Comment (mspang) [inline src/system/WatchableEventManagerSelect.cpp:89]: I'm not sure if I agree with this one being able to fail either..

If it fails, and all of the possible failures are either catastrophic faults or failed preconditions, do we really need a return value versus adding an assertion?

File: src/system/WatchableEventManagerSelect.h (hunk 2 above)
Comment (kpschoedel) [reply src/system/WatchableEventManagerSelect.h:46]: I'll make `Signal()` succeed or die.

`Init()` and `Shutdown()` may return implementation-dependent errors, just like the `System::Layer` versions. `WatchableEventManager` wants to be `SystemLayerImpl` when it grows up.

**Review by mspang — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**